src/node_internals.h

```diff
@@ -172,7 +172,7 @@ inline MUST_USE_RESULT bool ParseArrayIndex(v8::Local<v8::Value> arg,
     return true;
   }
 
-  int32_t tmp_i = arg->Int32Value();
+  int32_t tmp_i = arg->Uint32Value();
```

Comment:
yeah. and the changes you've made so far are excellent and I don't want to hold them back while we discuss how to handle this case.